### PR TITLE
File attachment doc improvements

### PR DESCRIPTION
**Pull request by snide**
## Summary

Reorgs and adds to the documentation around File attachments. Makes the following changes:

- Breaks up the content into the following
  - `concepts/file-attachments` for the big picture
  - `sdk/file-attachments` for the implementation details
  - `sdk/image-transformations` for the image specific implementation detail
  - `concepts/data-model` for the data-shape of the new column types (no changes here, just removed duplication in other pages)
- Added examples of transformations (requires https://github.com/xataio/website/pull/557)
- Provided examples of access URLs code
- Provided more examples of transformation code
- Added imagery throughout

### Remaining work

The following work still needs to be completed:

- [ ] Do a pass on any language changes with @joan-ing 
- [x] Get feedback on content break up with @philkra @joan-ing 
- [x] Finish adding TS examples for transformations
  - [x] Ask @philkra to fill in python examples for transformations (Some shapes in TS were different than what the URL pattern assumed, so I think we should give examples)
- [x] [Verify broken transformations](https://xata-hq.slack.com/archives/C04NJ5571GE/p1693278261341319)
- [ ] ~Replace hard-coded URLs with MDX consts~ For another PR
- [ ] ~Add example for a form upload~ for another PR
- [ ] ~Add [xata-gallery](https://xata-gallery.vercel.app) example references~ For another PR

### Dependencies

- [x] https://github.com/xataio/website/pull/557 needs to merge first

---

### Timing

**Release**:

- [ ] Hold for release - Till @snide approves

### CODE REVIEW

File: 040-SDK/160-image-transformations.mdx [hunk 1]
```diff
@@ -0,0 +1,881 @@
...
+
+```ts
+// Get a specific record from the "gallery" database
+const record = xata.db.gallery.read('record_id');
+
+// Get a new url and signedUrl for an image in the "imageColumnName"
+// This will create a square image that is cropped from the top
+const { url, signedUrl } = record.imageColumnName.transform({
+  height: 100,
+  rotate: 180
```
Comment (alexfrancoeur) [inline 040-SDK/160-image-transformations.mdx:37]: I noticed this was missing a comma in TS examples

File: 040-SDK/160-image-transformations.mdx [hunk 2]
```diff
@@ -0,0 +1,881 @@
...
+  options={{ rotate: 180, height: 100 }}
+/>
+
+The below example shows a method to transform an image directly off the file object in a record.
+
+<TabbedCode tabs={['TypeScript', 'Python', 'URL']}>
+
+```ts
+// Get a specific record from the "gallery" database
+const record = xata.db.gallery.read('record_id');
```
Comment (alexfrancoeur) [inline 040-SDK/160-image-transformations.mdx:31]: I also think this may need `await`

Comment (snide) [reply 040-SDK/160-image-transformations.mdx]: Well look at Mr. Engineer over here! Nice spot.

File: 040-SDK/160-image-transformations.mdx [hunk 3]
```diff
@@ -0,0 +1,916 @@
...
+
+```ts
+const transformedImageUrl = transformImage('https://us-east-1.storage.xata.sh/4u1fh2o6p10blbutjnphcste94', {
+  blur: 20
+});
+```
+
+```python
+img = xata.files().transform("https://us-east-1.storage.xata.sh/4u1fh2o6p10blbutjnphcste94", {
+  "background": 20
```
[the remark this answers is not in the document]
Comment (philkra) [reply 040-SDK/160-image-transformations.mdx:145]: yes, good spot! thank you. fixing it asap

[the remark this answers is not in the document]
Comment (philkra) [reply 040-SDK/160-image-transformations.mdx]: fixed with https://github.com/xataio/mdx-docs/pull/97/commits/347e85e3166787976a53ca6ec35e6c8d02e0f2be

**Review by philkra — APPROVED** [review]
LGTM, new structure, and code snippets look great

**Pull request merged**